backbone.js

```diff
@@ -38,6 +38,13 @@
   var array = [];
   var slice = array.slice;
 
+  // Optimized deep comparison function.
+  var _isEqual = _.isEqual;
+
+  function isEqual( a, b ){
+      return a && b && typeof a === 'object' && typeof b === 'object' ? _isEqual( a, b ) : a === b;
```

Comment:
This won't work for `0 === -0` or `NaN`. I expect this to be faster for cases where `_isEqual(a, b)` is false but roughly the same when `a === b` (minus JIT optimizations)